src/Bicep.LangServer/Handlers/BicepTextDocumentSyncHandler.cs

```diff
@@ -63,8 +76,19 @@ public override Task<Unit> Handle(DidOpenTextDocumentParams request, Cancellatio
             {
                 bicepConfigChangeHandler.HandleBicepConfigOpenEvent(documentUri);
             }
+
             //new compliation manager 
-            this.compilationManager.UpsertCompilation(documentUri, request.TextDocument.Version, request.TextDocument.Text, request.TextDocument.LanguageId);
+            var fileExtension = Path.GetExtension(request.TextDocument.Uri.Path);
+
+            if (fileExtension == ".bicep")
```

Comment:
This can be replaced with [`PathHelper.HasExtension`](https://github.com/Azure/bicep/blob/main/src/Bicep.Core/FileSystem/PathHelper.cs#L157).

Reply:
It would also be better to replace the hardcoded file extensions with `LanguageConstants.*FileExtension`.

Reply:
Looks like [PathHelper.HasExtension](https://github.com/Azure/bicep/blob/5048ab65c0c20a0f917fa0293d329f3e92a81e97/src/Bicep.Core/FileSystem/PathHelper.cs#L157) takes in System.Uri but in the handler method I am getting
 OmniSharp.Extensions.LanguageServer.Protocol.DocumentUri leading me to do an explicit cast like this:

<img width="592" alt="has_extension" src="https://user-images.githubusercontent.com/27924869/174844226-427565b9-fa63-4eb9-94aa-3894f65d1041.PNG">

Reply:
would this be good practice?

Reply:
I would recommend calling `request.TextDocument.Uri.ToUri()` instead of using a cast. The DocumentUri class does define how that cast should behave ([it calls `.ToUri()`](https://github.com/OmniSharp/csharp-language-server-protocol/blob/ce5dadc8fcaf9ea7aefbbe90868be891727bd57e/src/Protocol/DocumentUri.cs#L224-L233)), but I can never tell at a glance if a cast is invoking a defined `explicit operator` method or if it's doing something potentially unsafe.